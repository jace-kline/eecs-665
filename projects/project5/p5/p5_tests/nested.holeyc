int a;
bool b;
boolptr bptr;

bool g(int a) {
    return (a > 0);
}

bool func(boolptr bptr, int a) {
    a = a + 1;
    bool x;
    x = g(a);
    if (a > 10) {
        return x;
    }
    return func(func(^x,a), a);
}