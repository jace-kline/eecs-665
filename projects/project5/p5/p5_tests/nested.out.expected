int a;
bool b;
boolptr bptr;
bool g(int a){
	return a(int) > 0;
}
bool func(boolptr bptr, int a){
	a(int) = (a(int) + 1);
	bool x;
	x(bool) = (g(int->bool)(a(int)));
	if (a(int) > 10){
		return x(bool);
	}
	return func(boolptr,int->bool)(func(boolptr,int->bool)(^x(bool), a(int)), a(int));
}